src/modules/Dropdown/Dropdown.js

```diff
@@ -686,10 +702,12 @@ export default class Dropdown extends Component {
       search && searchQuery && 'filtered'
     )
     let _text = placeholder
-    if (text && !searchQuery) {
-      _text = text
-    } else if (searchQuery) {
+    if (searchQuery) {
       _text = null
+    } else if (text) {
+      _text = text
+    } else if (open && !multiple) {
+      _text = this.getSelectedItemText()
```

Comment:
After looking at the code closer, I see that it is correctly named and performing a different task.  I think we should refactor these two methods into one, however.  There is a lot of code duplication between them:

```js
getSelectedItemValue = () => {
  const { selectedIndex } = this.state
  const options = this.getMenuOptions()

  return _.get(options, `[${selectedIndex}].value`)
}

getSelectedItemText = () => {
  const { selectedIndex } = this.state
  const options = this.getMenuOptions()

  return _.get(options, `[${selectedIndex}].text`)
}
```

How about we replace it with:

```js
getSelectedItem = () => {
  const { selectedIndex } = this.state
  const options = this.getMenuOptions()

  return options[selectedIndex]
}
```

And the usages like so:

```diff
-const value = this.getSelectedItemValue()
+const value = _.get(this.getSelectedItem(), 'value')

-_text = this.getSelectedItemText()
+_text = _.get(this.getSelectedItem(), 'text')
```